src/reducers/listReducer.js

```diff
@@ -38,6 +38,21 @@ const sort = (list, sortValue) => {
   return list ? _.sortBy(list, sortValue) : list
 }
 
+const transform = (list, transforms) => {
+  let newList = list
+  for (let transform in transforms) {
+    const transformFunction = transforms[transform].func
+    if (transform === 'sort') {
```

Comment:
should be able to leave all this to the caller. That is, it is their job to coordinate the sorting and the filtering if any. There could be arbitrary many different transformations going on. The essence of this code moves to PageDefinition where we coordinate the filter and sort settings changes to set the `transformer` on the reducer.

Note that `transformer` would be a function that takes the list and produces a list. It can do arbitrary computation to get the final result.